src/components/include/application_manager/application_manager.h

```diff
@@ -292,6 +292,8 @@ class ApplicationManager {
    */
   virtual bool IsAppSubscribedForWayPoints(ApplicationSharedPtr app) const = 0;
 
+  virtual bool IsAppSubscribedForWayPoints(Application& app) const = 0;
```

Comment:
@mked-luxoft not clear why you added additional implementation without pointer. Is it possible to remove old one in that case? Confusing to have such public API.